x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -15,13 +15,42 @@ import { AdvancedPolicySchema } from '../models/advanced_policy_schema';
 
 function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
   let newPolicyConfig = obj;
+
+  // First set the value.
   for (let i = 0; i < path.length - 1; i++) {
     if (!newPolicyConfig[path[i]]) {
       newPolicyConfig[path[i]] = {} as Record<string, unknown>;
     }
     newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
   }
   newPolicyConfig[path[path.length - 1]] = value;
+
+  // Then, if the user is deleting the value, then we need to ensure we clean up the config.
+  // We delete any sections are the empty, whether that be an empty string,  empty object, or undefined.
+  if (value === '' || value === undefined) {
+    newPolicyConfig = obj;
+    for (let k = path.length; k >= 0; k--) {
+      const nextPath = path.slice(0, k);
+      for (let i = 0; i < nextPath.length - 1; i++) {
+        // Traverse and find the next section
+        newPolicyConfig = newPolicyConfig[nextPath[i]] as Record<string, unknown>;
+      }
+      if (
+        newPolicyConfig[nextPath[nextPath.length - 1]] === undefined ||
+        newPolicyConfig[nextPath[nextPath.length - 1]] === '' ||
+        Object.keys(newPolicyConfig[nextPath[nextPath.length - 1]] as object).length === 0
```

Comment:
I'm assuming this casting here is safe at runtime? will it always be an object if defined at this point?

Reply:
correct

Reply:
'as object' ? Shouldn't it be 'as Object' ?

Reply:
I think both work.  The type error I had on this went away after I added the cast